clients/payment/src/pages/PaymentPage.tsx

```diff
@@ -1,50 +1,481 @@
+import { LakeTextInput } from "@swan-io/lake/src/components/LakeTextInput";
+import { SegmentedControl } from "@swan-io/lake/src/components/SegmentedControl";
+import { Space } from "@swan-io/lake/src/components/Space";
+import { SwanLogo } from "@swan-io/lake/src/components/SwanLogo";
+import { backgroundColor, colors, spacings } from "@swan-io/lake/src/constants/design";
+import { useResponsive } from "@swan-io/lake/src/hooks/useResponsive";
+import { useUrqlMutation } from "@swan-io/lake/src/hooks/useUrqlMutation";
+import { isNotNullish } from "@swan-io/lake/src/utils/nullish";
+import { filterRejectionsToResult } from "@swan-io/lake/src/utils/urql";
+import { CountryPicker } from "@swan-io/shared-business/src/components/CountryPicker";
+import { CountryCCA3, allCountries } from "@swan-io/shared-business/src/constants/countries";
+import { validateRequired } from "@swan-io/shared-business/src/utils/validation";
+import { CSSProperties, useMemo } from "react";
+import { ScrollView, StyleSheet } from "react-native";
+import { combineValidators, hasDefinedKeys, useForm } from "react-ux-form";
+import { formatCurrency } from "../../../banking/src/utils/i18n";
+import { validateIban } from "../../../banking/src/utils/iban";
+import { SepaLogo } from "../components/SepaLogo";
+import {
+  AddSepaDirectDebitPaymentMandateFromPaymentLinkDocument,
+  GetMerchantPaymentLinkQuery,
+} from "../graphql/unauthenticated";
+import { setMandateUrl } from "../states/mandateUrl";
+import { languages, locale, setPreferredLanguage, t } from "../utils/i18n";
+import { Router } from "../utils/routes";
+
+const IMAGE_STYLE: CSSProperties = {
+  top: 0,
+  left: 0,
+  alignSelf: "center",
+  maxWidth: "200px",
+};
 
 const styles = StyleSheet.create({
-  base: {
-    backgroundColor: colors.gray[50],
-    flexGrow: 1,
+  container: {
+    marginHorizontal: "auto",
+    maxWidth: 960,
+    width: "100%",
+    paddingHorizontal: spacings[24],
+    paddingVertical: spacings[32],
+  },
+  label: {
+    flexBasis: "50%",
+    flexShrink: 1,
+    alignSelf: "stretch",
+  },
+  segmentedControlDesktop: {
+    maxWidth: "50%",
+  },
+  segmentedControl: {
+    maxWidth: "100%",
+  },
+  swanLogo: {
+    display: "inline-flex",
+    height: 9,
+    width: 45 * (9 / 10),
+  },
+  buttonItem: { width: "40%", paddingHorizontal: 0 },
+  buttonItemDesktop: { width: "15%" },
+  selectLanguage: {
+    alignItems: "flex-end",
   },
 });
 
-export const PaymentPage = () => {
-  const { data } = useUrqlQuery(
-    {
-      query: GetMerchantPaymentLinkDocument,
+type FormState = {
+  paymentMethod: "SepaDirectDebitCore";
+  iban: string;
+  country: CountryCCA3;
+  name: string;
+  addressLine1: string;
+  addressLine2: string;
+  city: string;
+  postalCode: string;
+  state: string;
+};
+
+type Props = {
+  paymentLink: NonNullable<GetMerchantPaymentLinkQuery["merchantPaymentLink"]>;
+};
+
+const sepaNonEeaCountries = ["CHE", "VAT", "GBR", "MCO", "SMR", "AND"];
```

Comment:
nit: could be nice to have the API provide those for us, hardcoding isn't ideal

Reply:
I spoke to the backend team about this